src/main/java/teammates/logic/api/Logic.java

```diff
@@ -1022,6 +1022,17 @@ public void deleteStudent(String courseId, String studentEmail) {
         studentsLogic.deleteStudentCascade(courseId, studentEmail);
     }
 
+    /**
+     * Deletes all the students in the course.
+     *
+     * @param courseId course id for the students
+     */
+    public void deleteAllStudentsInCourse(String courseId) {
+
+        Assumption.assertNotNull(courseId);
+        studentsLogic.deleteAllStudentsInCourse(courseId);
```

Comment:
Yes; one that does the cascade then calls `studentsDb.deleteStudentsForCourse(courseId);` as the final step. Not sure if there is an existing efficient way to do the cascade; maybe @shubham49 you can do a survey of the currently available logic and db methods to check if it's possible.